src/libraries/Microsoft.Extensions.Configuration.Xml/tests/XmlConfigurationTest.cs

```diff
@@ -193,6 +219,125 @@ public void NameAttributeInRootElementContributesToPrefix()
+        }
+
+        [Fact]
+        public void RepeatedElementsWithSameNameContributeToPrefix()
+        {
+            var xml =
+              @"<settings>
+                  <DefaultConnection Name='Data'>
+                      <ConnectionString>TestConnectionString1</ConnectionString>
+                      <Provider>SqlClient1</Provider>
+                  </DefaultConnection>
+                  <DefaultConnection Name='Data'>
+                      <ConnectionString>TestConnectionString2</ConnectionString>
+                      <Provider>SqlClient2</Provider>
+                  </DefaultConnection>
+                </settings>";
+            var xmlConfigSrc = new XmlConfigurationProvider(new XmlConfigurationSource());
+
+            xmlConfigSrc.Load(TestStreamHelpers.StringToStream(xml));
+
+            Assert.Equal("TestConnectionString1", xmlConfigSrc.Get("DefaultConnection:Data:0:ConnectionString"));
+            Assert.Equal("SqlClient1", xmlConfigSrc.Get("DefaultConnection:Data:0:Provider"));
+            Assert.Equal("TestConnectionString2", xmlConfigSrc.Get("DefaultConnection:Data:1:ConnectionString"));
+            Assert.Equal("SqlClient2", xmlConfigSrc.Get("DefaultConnection:Data:1:Provider"));
+        }
+
+        [Fact]
+        public void RepeatedElementsWithDifferentNamesContributeToPrefix()
+        {
+            var xml =
+              @"<settings>
+                  <DefaultConnection Name='Data1'>
+                      <ConnectionString>TestConnectionString1</ConnectionString>
+                      <Provider>SqlClient1</Provider>
+                  </DefaultConnection>
+                  <DefaultConnection Name='Data2'>
+                      <ConnectionString>TestConnectionString2</ConnectionString>
+                      <Provider>SqlClient2</Provider>
+                  </DefaultConnection>
+                </settings>";
+            var xmlConfigSrc = new XmlConfigurationProvider(new XmlConfigurationSource());
+
+            xmlConfigSrc.Load(TestStreamHelpers.StringToStream(xml));
+
+            Assert.Equal("TestConnectionString1", xmlConfigSrc.Get("DefaultConnection:Data1:ConnectionString"));
+            Assert.Equal("SqlClient1", xmlConfigSrc.Get("DefaultConnection:Data1:Provider"));
+            Assert.Equal("TestConnectionString2", xmlConfigSrc.Get("DefaultConnection:Data2:ConnectionString"));
+            Assert.Equal("SqlClient2", xmlConfigSrc.Get("DefaultConnection:Data2:Provider"));
+        }
+
+        [Fact]
+        public void NestedRepeatedElementsContributeToPrefix()
+        {
+            var xml =
+              @"<settings>
+                  <DefaultConnection>
+                      <ConnectionString>TestConnectionString1</ConnectionString>
+                      <ConnectionString>TestConnectionString2</ConnectionString>
+                  </DefaultConnection>
+                  <DefaultConnection>
```

Comment:
What would happen if I have:
```xml
<settings>
    <DefaultConnection Name='Data'>
        <ConnectionString>TestConnectionString1</ConnectionString>
        <Provider>SqlClient1</Provider>
    </DefaultConnection>
    <DefaultConnection Name='Data'>
        <ConnectionString>TestConnectionString2</ConnectionString>
        <Provider>SqlClient2</Provider>
    </DefaultConnection>
    <OtherValue>
        <Value>MyValue</Value>
    </OtherValue>
    <DefaultConnection>
        <ConnectionString>TestConnectionString3</ConnectionString>
        <Provider>SqlClient3</Provider>
    </DefaultConnection>
</settings>
```

Would I get that configuration section with: `xmlConfigSrc.Get("DefaultConnection:2:ConnectionString")` or with `xmlConfigSrc.Get("DefaultConnection:3:ConnectionString")` ? Is the index the index for the element, or is the index based on the number of elements with the same key name?

Reply:
For this example it would return `TestConnectionString3`. The following logic added in the src:

```c#
                            // check if this element has appeared before
                            XmlConfigurationElement sibling = allElements.Where(e => e.IsSibling(element)).OrderByDescending(e => e.Index).FirstOrDefault();
                            if (sibling != null)
                            {
                                sibling.Multiple = element.Multiple = true;
                                element.Index = sibling.Index + 1;
                            }
```
keeps track of repeated elements with the same key, with Multiple/Index values having default values of false/0, and elements with the same key name increment the index.

Reply:
@safern 

@maryamariyan is correct, it returns indices in line with what the equivalent JSON would produce. The index is based on elements with the same key. So in your example, you would need `xmlConfigSrc.Get("DefaultConnection:2:ConnectionString")`